### DISCUSSION TITLE
Documentation for adding a new backend ...

### DISCUSSION THREAD

**Discussion by radharamanan** [fastmachinelearning/hls4ml#1024] [Q&A]
Hi,

If I wish to add a new backend for a new HLS tool, is there a document that describes the steps necessary to do this? Also, do I need to work with anyone to attempt this?

**Comment by vloncar** [comment]
We don't have documentation on this as it is not something people often do, but it is certainly possible. At the base level, you need to subclass the `Backend` class and implement a few methods. Depending on the type of a backend you wish to implement you may want to share the code with existing backends instead of doing everything manually. What kind of a backend do you want to implement?

**Reply by radharamanan** [reply]
Is there a way to communicate over email? I can be more explicit in direct communication instead of in a forum.

**Reply by vloncar** [reply]
I work at CERN, same username :wink: The emails use the older .ch domain.

**Reply by radharamanan** [reply]
Thanks. Emailed you.
